src/common/backed_args.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2025, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/container/inlined_vector.h>
+
+#include <string_view>
+
+namespace cmn {
+
+#ifdef ABSL_HAVE_ADDRESS_SANITIZER
+constexpr size_t kReqStorageSize = 44;
+#else
+constexpr size_t kReqStorageSize = 158;
+#endif
+
+struct BackedArguments {
+  BackedArguments() {
+  }
+
+  BackedArguments(BackedArguments&& o) noexcept {
+    *this = std::move(o);
+  }
+
+  BackedArguments& operator=(BackedArguments&& o) noexcept {
+    args = std::move(o.args);
+    storage = std::move(o.storage);
+    if (!args.empty()) {
+      const char* p = args.front().data();
+      bool is_inline = (p >= o.storage.begin() && p <= o.storage.end());
+      if (is_inline) {  // fix references
+        char* next = storage.data();
+        for (auto& arg : args) {
+          arg = std::string_view{next, arg.size()};
+          next += (arg.size() + 1);  //  + \0
+        }
+      }
+    }
+    return *this;
+  }
+
+  size_t StorageCapacity() const {
+    return args.capacity() + storage.capacity();
+  }
+
+  // The capacity is chosen so that we allocate a fully utilized (256 bytes) block.
+  using StorageType = absl::InlinedVector<char, kReqStorageSize>;
+
+  absl::InlinedVector<std::string_view, 5> args;
```

Comment:
You can use a `InlinedVector<uint32_t> sizes` to save 75% memory. Sure, it requires decoding time, but you loose it now as well by fixing references in moves (we move at least twice)